src/notification.rs

```diff
@@ -10,68 +15,132 @@ pub use self::toast::prepare;
 
 mod toast;
 
-pub fn connected(config: &config::Notifications<Sound>, server: impl ToString) {
-    show_notification(&config.connected, "Connected", server);
-}
-
-pub fn reconnected(config: &config::Notifications<Sound>, server: impl ToString) {
-    show_notification(&config.reconnected, "Reconnected", server);
+pub struct Notifications {
+    notified_times: HashMap<Notification, DateTime<Utc>>,
 }
 
-pub fn disconnected(config: &config::Notifications<Sound>, server: impl ToString) {
-    show_notification(&config.disconnected, "Disconnected", server);
-}
+impl Notifications {
+    pub fn new() -> Self {
+        Self {
+            notified_times: HashMap::new(),
```

Comment:
Nit: "notified_times" is a bit confusing imo 🤔 
Perhaps something like:
- notification_timestamps
- recent_notifications 
or something similar?

Reply:
Yeah I had went through a few iterations of what to name this. I am not attached to `notified_times`. I was going to use `notified_at` at some point, but I think maybe `recent_notifications` makes most sense.

I've went ahead and made the change. Thank you!